### ISSUE TITLE
Vala documentation

### ISSUE THREAD

**Issue by tintou** [hotdoc/hotdoc#148]
It should be able to produce Vala documentation with the GObject Introspection. In many libraries, Vala has a .metadata file that overrides some annotations from the .gir in order to have an usable API

**Comment by MathieuDuponchelle** [comment]
That would be nice indeed. The work needed there is:

- [ ] updating the gi formatter in order to translate symbols to the Vala syntax, much like we do for eg `python`
- [ ] Parsing vala-specific files to enrich / modify the newly-added Vala representation.

Weren't there also `.vapi` files, or am I remembering wrong?

**Comment by tintou** [comment]
Yes, there are foo-custom.vapi files to add more symbols (often adding macros to Vala when the original library comes from C)

**Comment by MathieuDuponchelle** [comment]
Patches welcome, vala isn't my language of choice but I'll be happy to merge support for it :) Regarding the addition of new symbols, a new mechanism will be needed I think, that can be used for python overrides as well
